active/NumberBox/NumberBox.md

```diff
@@ -159,24 +185,31 @@ unsealed runtimeclass NumberBox : Windows.UI.Xaml.Controls.Control
     String Text; 
     String PlaceholderText;
```

Comment:
I'm not sure that PlaceholderText (although I bind it to TextBox.PlaceholderText) will ever be visible. A community member has an open question about whether the NumberBox can ever be empty -- right now it can't.

Reply:
Is that an artifact of needing a valid digit occupying the Text field?

Reply:
Yes, the text should always display whatever is in the Value property. @adrientetar would like the text to be empty when the value is NaN, shall we do that?

Reply:
We've moved away from NaN as a sentinel value and using nullable instead.

Reply:
Should I make Value an IReference<double> then?

Reply:
@adrientetar could you help me understand the value of being able to have an empty NumberBox? 

In order to reduce end user confusion, the guidance is that a NumberBox should have _at least_ either a numeric value or placeholder text indicating how a NumberBox should be used (e.g, "A + B"). 

The only reason I see for allowing nullable is so that PlaceholderText can be visible - but this adds complexity to the validation features.  

I'm interested to hear @adrientetar's perspective - otherwise nullable + PlaceholderText could be left for V2 pending justifiable interest.

Reply:
@SavoySchuler, I have number boxes that display the position and size of selected objects in my app (and you can edit to move, similar to Adobe XD sidebar fields). When nothing is selected, these get disabled and display nothing. So yes, being able to display nothing is important.

Reply:
Could that be accomplished by having no placeholder text?

Reply:
No, when the control doesn't have focus it always displays the current Value, which is either a number or NaN.

Reply:
To synthesize, @adrientetar is using the nullable state as an indicator to disable the corresponding attribute. I certainly see this need scaling. 

@teaP Is a valid nullable state an easy add?

Reply:
Shouldn't be too bad. The downside from the app side is I don't believe you can x:Bind to a normal double, and prior to RS5 you couldn't set the value in xaml at all.

Reply:
@jevansaks do you have insight on this? 

Those cons seem unacceptably large. A workaround for scenarios like @adrientetar's could be to disable the corresponding property outside of the control with a CheckBox. While this isn't as seamless as what @adrientetar has implemented, the the pro does not seem worth the cost of losing x:Bind and requiring RS5 for full compatibility. 

 scenario  to handle disabling the property

Reply:
I mean, if you add a way to display NaN as empty string (maybe optionally), that's all I really need. Or could I somehow set the Text property in this case to empty string?

Reply:
With the nullable, you could set the Value to null. That would cause it to display the PlaceholderText (which could be empty).

Reply:
> The downside from the app side is I don't believe you can x:Bind to a normal double,

Yes but you make a double? property in your view model then that property can be assigned double values seamlessly, IIUC.

As an alternative… Qt's QSpinBox has protected functions to override the conversion between value and text

![image](https://user-images.githubusercontent.com/1250894/69162634-d521dc80-0aa1-11ea-9481-e98ef7667edc.png)

If there is something like that, I could use to change conversion of NaN to empty string. I mean that would suffice for my use, nullable is still good if you want to be able to display null state and NaN separately.

Reply:
@adrientetar - we have enabled this scenario. When a NumberBox is empty, the value will read NaN and the placeholder text will display (if set).